src/Components/PaymentSuccess/PaymentSuccessView.js

```diff
@@ -30,3 +28,39 @@ export const PaymentSuccessView = ({ onClose }) => {
   }
   return null;
 };
+
+const getCurrentFlow = () => {
+  const { product, giftRecipient, giftCode } = usePelcro.getStore();
+
+  if (giftRecipient) {
+    return "giftCreate";
+  } else if (giftCode) {
+    return "giftRedeem";
+  } else if (product) {
+    return "subscriptionSuccess";
+  }
+};
+
+const getSuccessWording = (i18n) => {
+  const flow = getCurrentFlow();
+  const { product, giftRecipient } = usePelcro.getStore();
+
+  const wordingDictionary = {
+    subscriptionSuccess: {
```

Comment:
I think this is another situation where being able to pass data to `switchView` calls would've made the implementation a bit better